docs/src/upgrade/release-4.x.x.asciidoc

```diff
@@ -402,6 +402,12 @@ g.V().has_label(GValue('label', 'person'))
 Use of `GValue` in traversals with repeated patterns may lead to improved performance in certain graphs. Consult the
 documentation for your specific graph provider for recommendations on how to best utilize `GValue` in traversals.
 
+==== Change Default Script Language in Gremlin-Server
```

Comment:
the `language` field is one way, but seems worth mentioning that they could also set it from the server-side as well by switching the configuration back to gremlin-groovy in the yaml. that's another possible way, right?

Reply:
As it's currently implemented, if a request comes in with no specified language, the server assumes a language of `gremlin-lang` (see changes to [HttpGremlinEndpointHandler](https://github.com/apache/tinkerpop/pull/2966/files#diff-ddd50660d38c3c8ba46087caed5ee89f059a3f760e951081e75d6f5864616208) and [HttpRequestMessageDecoder](https://github.com/apache/tinkerpop/pull/2966/files#diff-ee912e4dd22cabd2776846b569646a94ebfbd99aa073eed2360d53ddef046724)).

It's currently not smart enough to switch this based on available script engines in the server.

I will reconsider the messaging in the upgrade docs and see if I can include some more nuance in it.

Reply:
since you have both configured explicitly now, i guess its more clear that gremlin-groovy is still there. i think the context is more clear now as you have it written.

minor thing, please consider more concise section title as longer ones never look nice in the toc navigation on the left. mabye "Gremlin Server Default Language" would be better?